src/primitive.c

```diff
@@ -219,18 +219,16 @@ Symmetry *prm_get_primitive_symmetry(double t_mat[3][3],
                                       prim_symmetry->trans[i]);
     }
 
-#ifdef SPGDEBUG
-    int j;
     for (i = 0; i < prim_symmetry->size; i++) {
-        fprintf(stderr, "--- %d ---\n", i + 1);
-        for (j = 0; j < 3; j++) {
-            fprintf(stderr, "%d %d %d\n", prim_symmetry->rot[i][j][0],
-                    prim_symmetry->rot[i][j][1], prim_symmetry->rot[i][j][2]);
+        debug_print("--- %d ---\n", i + 1);
+        for (int j = 0; j < 3; j++) {
+            debug_print("%d %d %d\n", prim_symmetry->rot[i][j][0],
+                        prim_symmetry->rot[i][j][1],
+                        prim_symmetry->rot[i][j][2]);
         }
-        fprintf(stderr, "%f %f %f\n", prim_symmetry->trans[i][0],
-                prim_symmetry->trans[i][1], prim_symmetry->trans[i][2]);
+        debug_print("%f %f %f\n", prim_symmetry->trans[i][0],
+                    prim_symmetry->trans[i][1], prim_symmetry->trans[i][2]);
     }
```

Comment:
I re-read your initial PR comments. Probably your advice `Good coding practices these days is to avoid #ifdef` can be applied here, too.

Reply:
I am just curious when these redundant lines are promised to be omitted, -O2 and -O3? Is there a reference for that?

Reply:
Ok Here we are: https://godbolt.org/z/xqE9rhq5b
So as little as `-O1`

Reply:
> I re-read your initial PR comments. Probably your advice "Good coding practices these days is to avoid `#ifdef`" can be applied here, too.

Indeed, I very much tried to think of a way around it, but I couldn't find a way to make the printing conditional. I did consider getting rid of those variable and using `NDEBUG`, but it didn't seem compatible with how `warning_print` is used. That's why I've made them into separate cmake options

Reply:
Thank you for your detailed explanation. It's interesting that even O1 completely optimizes the code.